### PR TITLE
Make state functions update in memory state too

### PR DESCRIPTION
**Pull request by jodh-intel**

### CODE REVIEW

File: pod.go [hunk 1]
```diff
@@ -833,6 +854,20 @@ func (p *Pod) setPodState(state State) error {
 	return nil
 }
 
+func (p *Pod) getContainer(containerID string) (*Container, error) {
+	if containerID == "" {
+		return &Container{}, errNeedContainerID
+	}
+
+	for _, c := range p.containers {
+		if c.ID() == containerID {
```
Comment (sboeuf) [inline pod.go]: You don't need the getter here because we are in the virtcontainers lib itself, you can do `c.id` instead of `c.ID()`.

**Comment by sboeuf** [comment]
@jodh-intel I like the PR, it makes things clearer. Have you tested that with our runtime to make sure things are still working and it does not introduce a regression. I don't think it would but it always better to check !

**Comment by sameo** [comment]
LGTM

**Comment by jodh-intel** [comment]
Hi @sboeuf - yes, tested with https://github.com/clearcontainers/runtime/commit/66b54638c5125167fe852dce4a45c2a1d01c9e27.

**Comment by sboeuf** [comment]
LGTM

**Pull request merged**